### PR TITLE
Fix: avoid double-rendering sql_header in dbt show

### LINKED ISSUE #8739: [CT-3170] [Fix] sql_header is being rendered twice in dbt show
Root cause: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/adapters/show.sql#L2-L6

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves #8739

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: tests/adapter/dbt/tests/adapter/dbt_show/test_dbt_show.py [hunk 1]
```diff
@@ -50,7 +50,6 @@ def models(self):
         }
 
     def test_sql_header(self, project):
-        run_dbt(["build", "--vars", "timezone: Asia/Kolkata"])
```
Comment (MichelleArk) [inline tests/adapter/dbt/tests/adapter/dbt_show/test_dbt_show.py:53]: this was not necessary

**Review by colin-rogers-dbt — APPROVED** [review]

**Review by McKnight-42 — APPROVED** [review]

**Comment by MichelleArk** [comment]
Updated BQ test: https://github.com/dbt-labs/dbt-bigquery/pull/927

**Comment by MichelleArk** [comment]
Updated redshift test: https://github.com/dbt-labs/dbt-redshift/pull/611

**Comment by MichelleArk** [comment]
Updated snowflake test: https://github.com/dbt-labs/dbt-snowflake/pull/779

**Pull request merged**